tiledb/sm/serialization/array_schema.cc

```diff
@@ -1802,6 +1802,212 @@ Status max_buffer_sizes_deserialize(
   return Status::Ok();
 }
 
+void load_array_schema_request_to_capnp(
+    capnp::LoadArraySchemaRequest::Builder& builder,
+    const Config& config,
+    const LoadArraySchemaRequest& req) {
+  auto config_builder = builder.initConfig();
+  throw_if_not_ok(config_to_capnp(config, &config_builder));
+  builder.setIncludeEnumerations(req.include_enumerations());
+}
+
+void serialize_load_array_schema_request(
+    const Config& config,
+    const LoadArraySchemaRequest& req,
+    SerializationType serialization_type,
+    Buffer& data) {
+  try {
+    ::capnp::MallocMessageBuilder message;
+    auto builder = message.initRoot<capnp::LoadArraySchemaRequest>();
+    load_array_schema_request_to_capnp(builder, config, req);
+
+    data.reset_size();
+    data.reset_offset();
+
+    switch (serialization_type) {
```

Comment:
I see a lot of these switch cases in the serialization code... Is this something we could generalize? If so, let's file a follow up story I don't want to block this PR for this.

Reply:
I copied this from the prior art in other places. I do think this whole function could probably be turned into a template though since they're pretty boiler plate without a lot of changes for different message types.